src/azul/drs.py

```diff
@@ -36,30 +40,28 @@ def object_url(file_uuid: str,
 
 def http_object_url(file_uuid: str,
                     file_version: Optional[str] = None,
-                    base_url: Optional[str] = None) -> str:
+                    base_url: Optional[str] = None,
+                    access_id: Optional[str] = None) -> str:
     """
-    The HTTP(S) URL for a given DSS file UUID and version. The return value will point at the bare-bones DRS data
-    object endpoint in the web service.
+    The HTTP(S) URL for a given DSS file UUID and version. The return value will
+    point at the bare-bones DRS data object endpoint in the web service.
 
     :param file_uuid: the DSS file UUID of the file
-
     :param file_version: the DSS file version of the file
-
-    :param base_url: an optional service endpoint, e.g for local test servers. If absent, the service endpoint for the
-                     current deployment will be used.
+    :param base_url: an optional service endpoint, e.g for local test servers.
+                     If absent, the service endpoint for the current deployment
+                     will be used.
+    :param access_id: access id will be included in the URL if this parameter is
+                      supplied
     """
     scheme, netloc = _endpoint(base_url)
     return urlunsplit(SplitResult(scheme=scheme,
                                   netloc=netloc,
-                                  path=drs_http_object_path(file_uuid),
+                                  path='/'.join((drs_url, file_uuid, *(('access', access_id) if access_id else ()))),
```

Comment:
Why'd you rip out the nicely abstracted URL composition helper? At first glance, this looks worse than before.

Reply:
I thought that the abstraction was more verbose and obfuscating. Using it for the route of the `objects/{uuid}/access/{access_id}` URL made the parameter names inaccurate. I could refactor it, but since it was just doing string concatenation, it was less code to remove it. To me the concatenation is just as or more clear.

Reply:
Why don't you give the original approach another shot and then we can compare.